README.md

```diff
@@ -49,11 +63,52 @@ and installed from source manually, see the Developer guide for further instruct
     EOF
     ```
 
-2. Create [DSCInitializationc](#example-dscinitialization) CR manually.
+2. Create [DSCInitialization](#example-dscinitialization) CR manually.
   You can also use operator to create default DSCI CR by removing env variable DISABLE_DSC_CONFIG from CSV following restart operator pod.
 
 3. Create [DataScienceCluster](#example-datasciencecluster) CR to enable components
 
+### API overview
```

Comment:
How was this overview created? Is there a command or something that could be included in the Makefile or similar. Else I suspect this will be out of date before this commit hit the main branch..

Reply:
I had to do it manually 😓

Reply:
I guess whenever a new section in the API is created, it can be updated right. We can add it as a guideline in the contribution docs

Reply:
@AjayJagan Can you see a way to make it into a command? What's the source for the info?

Reply:
The source is the the CRDs generated in the manifests folder

Reply:
i would prefer not adding this section, unless it is auto updated, and user can always check from https://pkg.go.dev/github.com/opendatahub-io/opendatahub-operator/v2/apis/dscinitialization/v1#DSCInitializationSpec to read what to config.

Reply:
hmm so is it okay to remove this section?

Reply:
changed to the links as they are already auto generated.

Reply:
or you create a GH action, to run "kubectl explain dsci" and have the return in md format inject into README?

Reply:
but that would require us to create a cluster and deploy the cards in it right :(

Reply:
https://github.com/elastic/crd-ref-docs?tab=readme-ov-file ?

Reply:
hey @aslakknutsen I am working on script to automate the process. Will update the pr soon :)

Reply:
@AjayJagan isn't the tool from Elastic good enough? Do we want to maintain more scripts instead of using existing tools and potentially collaborate?

Reply:
hmm yea, that is a good point. Although I have finished the script 😛  I will try to use this lib then 👍

Reply:
@aslakknutsen @bartoszmajsak I have used the crd-ref-doc library. Let me know how it looks :)

Reply:
@AjayJagan When #894 is merged I'm assuming you'll get the Component level in there as well. 

Is there an option to exclude a package in the lib? Not sure "featuretracker" is useful here? @bartoszmajsak

Reply:
great!!
As for the exclusion of "featuretracker", yes we can do it.

Reply:
I think `FeatureTracker` is not intended to be user-facing API. It's internal mechanism for the operator to control what is installed or removed for a given piece of functionality. Making it as part of the user-facing docs might be confusing.

Reply:
Removed FeatureTracker from the doc

Reply:
@bartoszmajsak That could have been in /infrastructure/ ;)

Reply:
> @bartoszmajsak That could have been in /infrastructure/ ;)

I totally agree! Shall we move it there?

Reply:
@aslakknutsen @bartoszmajsak , after rebasing the code(#894), it doesnot update the docs with the content from `infrastructure/v1` and I think the reason is because it is missing the `groupversion_info.go` file

Reply:
@bartoszmajsak I'd hold of a little on it just because.. maybe, just maybe we'll include it in api with some upcoming feature rework.. hehe

Reply:
@AjayJagan hmm.. yeah, it's not really it's own group but part of DSCI.. What happens if you copy the groupversion_info.go from the dsci folder over to infrastructure as well?

@bartoszmajsak Was it perhaps intended that it could be included on both dsci and dsc level? It seems to only be dsci atm.

Reply:
@aslakknutsen Yep that works :D

Reply:
> Was it perhaps intended that it could be included on both dsci and dsc level? It seems to only be dsci atm.

@aslakknutsen it's used in both actually - e.g. 
https://github.com/opendatahub-io/opendatahub-operator/blob/927340ad089a3335a957f858f3d68560a4adedf6/components/kserve/kserve.go#L51

Reply:
in that case, how do we link it to 2 group versions 🤔